sites/all/modules/reol_app_feeds/src/Helper/NodeHelper.php

```diff
@@ -0,0 +1,313 @@
+    }
+
+    $values = $entity->{$field_name}[LANGUAGE_NONE];
+
+    if (NULL !== $sub_field_name) {
+      $values = array_column($values, $sub_field_name);
+    }
+
+    return $multiple ? $values : reset($values);
+  }
+
+  /**
+   * Get text value of a field.
+   *
+   * @param object $entity
+   *   The entity.
+   * @param string $field_name
+   *   The field names.
+   * @param string|null $sub_field_name
+   *   The optional sub-field names.
+   * @param bool $multiple
+   *   If set, multiple field values will be returned as an array. Otherwise,
+   *   only the first value will be returned.
+   *
+   * @return array|mixed|null
+   *   The text field value.
+   */
+  public function getTextFieldValue($entity, $field_name, $sub_field_name = NULL, $multiple = FALSE) {
+    $values = $this->getFieldValue($entity, $field_name, $sub_field_name, TRUE);
+    $values = array_map([$this, 'getTextValue'], $values);
+
+    return $multiple ? $values : reset($values);
+  }
+
+  /**
+   * Get list of paragraphs on a node.
+   *
+   * @param object $entity
+   *   The entity.
+   *
+   * @return array|\ParagraphsItemEntity[]
+   *   The paragraphs.
+   */
+  public function getParagraphs($entity) {
+    $paragraphs = [];
+
+    $helper = new ParagraphHelper();
+
+    $fields = $helper->getParagraphFields($entity);
+    foreach ($fields as $field_name => $field) {
+      $paragraphs += $helper->loadParagraphs($entity, $field_name);
+    }
+
+    return $paragraphs;
+  }
+
+  /**
+   * Get text value.
+   *
+   * @param mixed $value
+   *   The value.
+   *
+   * @return string|null
+   *   The text value if any.
+   */
+  private function getTextValue($value) {
+    return isset($value['safe_value']) ? $value['safe_value'] : NULL;
+  }
+
+  /**
+   * Get body text value from a node.
+   *
+   * @param object $node
+   *   The node.
+   *
+   * @return string|null
+   *   The body text value if any.
+   */
+  public function getBody($node) {
+    return $this->getTextFieldValue($node, 'body', NULL, FALSE);
+  }
+
+  /**
+   * Get image url.
+   *
+   * @param mixed $value
+   *   The value.
+   * @param bool $multiple
+   *   If set, multiple values will be returned as an array. Otherwise,
+   *   only the first value will be returned.
+   * @param string|null $image_style_name
+   *   An optional image style to apply if it exists.
+   *
+   * @return string[]|string|null
+   *   The image url(s).
+   */
+  public function getImage($value, $multiple = FALSE, $image_style_name = NULL) {
+    if (NULL === $value) {
+      return NULL;
+    }
+    if (isset($value[LANGUAGE_NONE])) {
+      $value = $value[LANGUAGE_NONE];
+    }
+    if (ParagraphHelper::isAssoc($value)) {
+      $value = [$value];
+    }
+    $uris = array_column($value, 'uri');
+    if (NULL !== $image_style_name && !empty(image_style_load($image_style_name))) {
+      $urls = array_map(function ($uri) use ($image_style_name) {
+        return image_style_url($image_style_name, $uri);
```

Comment:
It the image styles have been flushed and no-body have seen the page the image style for a given image is used yet. The file will not exists yet and will give an 404 on the URL.

This below is not tested and found on the internet :-)
```
$derivative_uri = image_style_path($style, $image_uri);
$success        = file_exists($derivative_uri) || image_style_create_derivative(image_style_load($style), $image_uri, $derivative_uri);
$new_image_url  = file_create_url($derivative_uri);
```

Reply:
Are you sure about this? In my tests the image is generated when requested.

Reply:
It might have changed. Just remember the issue form years back.